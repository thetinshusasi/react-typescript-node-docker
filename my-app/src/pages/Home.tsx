import React, { useEffect, useState, useContext } from "react";
import { Col, Container, Row } from "react-bootstrap";
import AddTodo from "../components/addToDo";
import AwaitLoading from "../components/awaitLoading";
import ToDoList from "../components/toDoList";

import Todos from "../containers/toDos";
import { getDoneTasks, getPendingTasks } from "../helpers/utils";
import { Status } from "../models/enums/status";
import IToDoItem from "../models/interfaces/todoItem";
import { TodosContext, TodosProviderType } from "../Providers/todosProvider";
const todolist: IToDoItem[] = [
  {
    _id: "1",
    title: "100",
    description: "desc1",
    status: Status.PENDING,
  },
];
export interface IHomeProps {}
export default function Home(props: IHomeProps) {
  const { loading, error, todos, getTodoList }: TodosProviderType = useContext(
    TodosContext
  );
  const [pendingTasks, setPendingTasks] = useState(getPendingTasks(todos));
  const [doneTasks, setDoneTasks] = useState(getDoneTasks(todos));

  useEffect(() => {
    getTodoList && getTodoList();
  }, [getTodoList]);
  useEffect(() => {
    setPendingTasks(getPendingTasks(todos));
    setDoneTasks(getDoneTasks(todos));
  }, [todos]);

  return (
    <React.Fragment>
      <Container fluid>
        <Row>
          <Col xs={12}>
            <AddTodo />
          </Col>
          <Col xs={12}>
            <Todos title="To Do Tasks">
              <AwaitLoading loading={loading} error={error}>
                <ToDoList
                  todos={pendingTasks}
                  message="Please add items in list by creating a task"
                />
              </AwaitLoading>
            </Todos>
          </Col>
          <Col xs={12}>
            <Todos title="Done Tasks">
              <AwaitLoading loading={loading} error={error}>
                <ToDoList
                  todos={doneTasks}
                  isComplete={true}
                  message="Please complete a task"
                />
              </AwaitLoading>
            </Todos>
          </Col>
        </Row>
      </Container>
    </React.Fragment>
  );
}
