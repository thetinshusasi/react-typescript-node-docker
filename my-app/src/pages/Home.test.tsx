import React from "react";
import { shallow } from "enzyme";
import Home from "./Home";
import AddTodo from "../components/addToDo";
import AwaitLoading from "../components/awaitLoading";
import ToDoList from "../components/toDoList";
import Todos from "../containers/todos";

describe("<Home />", () => {
  it("renders Home", () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
    expect(component.find(AddTodo).length).toEqual(1);
    expect(component.find(Todos).length).toEqual(2);
    expect(component.find(AwaitLoading).length).toEqual(2);
    expect(component.find(ToDoList).length).toEqual(2);
  });
});
