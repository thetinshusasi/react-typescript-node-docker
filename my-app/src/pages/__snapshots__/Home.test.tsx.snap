// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders Home 1`] = `
<Fragment>
  <Container
    fluid={true}
  >
    <Row
      noGutters={false}
    >
      <Col
        xs={12}
      >
        <AddTodo />
      </Col>
      <Col
        xs={12}
      >
        <ToDos
          title="To Do Tasks"
        >
          <AwaitLoading>
            <ToDoList
              message="Please add items in list by creating a task"
              todos={Array []}
            />
          </AwaitLoading>
        </ToDos>
      </Col>
      <Col
        xs={12}
      >
        <ToDos
          title="Done Tasks"
        >
          <AwaitLoading>
            <ToDoList
              isComplete={true}
              message="Please complete a task"
              todos={Array []}
            />
          </AwaitLoading>
        </ToDos>
      </Col>
    </Row>
  </Container>
</Fragment>
`;
